src/lib.rs

```diff
@@ -98,29 +125,50 @@ async fn proxy_handler(
         req.headers()
     );
 
+    // Currently only testing HTTP1 support
+    match req.version() {
+        Version::HTTP_2 => {
+            panic!("HTTP2 unsupported version")
+        }
+        Version::HTTP_3 => {
+            panic!("HTTP3 unsupported version")
+        }
+        _ => {}
+    }
+
```

Comment:
If only one version is supported, generally better to explicitly list it and then deny all others.
Otherwise this code would "support" HTTP_4, HTTP_5 and so on.